### PR TITLE
Rewrite docstring for denoise_tv_chambolle

### LINKED ISSUE #5379: clarify the meaning of the weights term for TV denoising functions
## Description
In the Zulip chat, there was [a question regarding the "weight" term](https://skimage.zulipchat.com/#narrow/stream/175598-general/topic/total.20variation.20documentation) in `denoise_tv_bregman`.

In that implementation, smaller weights lead to more smoothing whereas the opposite is true for `denoise_tv_chambolle`.

The underlying cause seems just to be which term in the cost function the weight applies to (i.e. either the data-consistency or the TV regularization term). We should update the docstrings to clarify where the weight is being applied to make its function less opaque.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

Closes #5379 by defining the `weight` parameter explicitly. 

I have reworked the docstrings for both `denoise_tv_chambolle` and `denoise_tv_bregman`; they now share this entire paragraph:
paragraph
and some parameter descriptions as well; they also refer to each other (with the "see also" subsection).

@grlee77 regarding the expression of the cost function, I have decided to go for the Getreuer convention because it is so standard in math (see also https://en.wikipedia.org/wiki/Total_variation_denoising#Rudin%E2%80%93Osher%E2%80%93Fatemi_PDE in spite of the other sections where lambda has a different meaning).

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Comment by mkcor** [comment]
I'll do a similar cleanup for the other function (`denoise_tv_bregman`).

**Comment by mkcor** [comment]
@grlee77 on the rendering side: I haven't used the math display style you recommended [here](https://github.com/scikit-image/scikit-image/pull/6134#issuecomment-1002758903), because this is how it renders now:
display_style

I noticed this on both Firefox and Chrome, when building the docs to check my changes, then I realized that we have this rendering issue throughout the docs wherever `..math::` is used. I guess something changed upstream, in Sphinx... :thinking:

It's as if `..math::` were ignored: Simply indenting a math formula produces the same rendering, as in
https://github.com/scikit-image/scikit-image/blob/721cb69dba1e73a65f297bc49ec680967ef8d443/skimage/restoration/_denoise.py#L451-L455
indented

So, for this PR, I have used the math inline style, so it renders as expected:
inline

**Comment by mkcor** [comment]
I guess I should escape the 'absolute value' symbol if I want to include it (ef68438e3aaf7a4c28990a1d71f7a58fe96bb7d5). Locally, the docs would still build for me and render like this:
abs_val

PS: I realize we don't the absolute value around $E_0$ since the cost function is non-negative, of course! But we need it around the difference $(E_{n-1} - E_n)$.

**Comment by grlee77** [comment]
> It's as if ..math:: were ignored: Simply indenting a math formula produces the same rendering, as in

Hmm, that is odd. That math directive is still described in current sphinx docs and no deprecation is mentioned.

**Comment by grlee77** [comment]
> I guess I should escape the 'absolute value' symbol if I want to include it 

Yeah, you have to either escape some of the symbols or use a "raw string" (change `"""` to `r"""` at the start of the docstring)

**Comment by mkcor** [comment]
> > It's as if ..math:: were ignored: Simply indenting a math formula produces the same rendering, as in
> 
> Hmm, that is odd. That math directive is still described in current sphinx docs and no deprecation is mentioned.

@lagru has figured it out: We need to stick with the proper syntax and add a space between `..` and `math`! :clap:

File: skimage/restoration/_denoise.py [hunk 1]
```diff
@@ -437,32 +437,41 @@ def _denoise_tv_chambolle_nd(image, weight=0.1, eps=2.e-4, max_num_iter=200):
 @utils.deprecate_multichannel_kwarg(multichannel_position=4)
 def denoise_tv_chambolle(image, weight=0.1, eps=2.e-4, max_num_iter=200,
                          multichannel=False, *, channel_axis=None):
-    """Perform total-variation denoising on n-dimensional images.
+    r"""Perform total variation denoising in nD.
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:440]: Oops, there was no `.. math::` directive nor `:math:` role here, so what was I thinking... :woman_facepalming: Fixed in 3e5c3b7629a168acccca291a231fef6a0bc741b6.

**Comment by mkcor** [comment]
I'm usually very keen on hyphenating so, as much as changing "an image with less total-variation" to "an image with less total variation" is a *correction*, changing "total-variation [substantive]" to "total variation [substantive]" is not: I made this change just for the sake of consistency with the references we point to, since they all write "total variation denoising."

**Review by lagru — COMMENTED** [review]
Thanks for working on this clear improvement. Take into consideration that I'm not very familiar with this so I'm struggling a little bit to verify the documented math. :wink:

File: skimage/restoration/_denoise.py [hunk 2]
```diff
@@ -307,16 +319,32 @@ def denoise_tv_bregman(image, weight=5.0, max_num_iter=100, eps=1e-3,
...
+    --------
+    denoise_tv_chambolle : Perform total variation denoising in nD.
+
     References
     ----------
-    .. [1] https://en.wikipedia.org/wiki/Total_variation_denoising
-    .. [2] Tom Goldstein and Stanley Osher, "The Split Bregman Method For L1
+    .. [1] Tom Goldstein and Stanley Osher, "The Split Bregman Method For L1
            Regularized Problems",
            ftp://ftp.math.ucla.edu/pub/camreport/cam08-29.pdf
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:342]: Sure! I used `wget` to retrieve the PDF over ftp (just like I would do with http/https), but I guess https makes it more accessible.

File: skimage/restoration/_denoise.py [hunk 3]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
         Otherwise, this parameter indicates which axis of the array corresponds
         to channels.
 
         .. versionadded:: 0.19
            ``channel_axis`` was added in 0.19.
     multichannel : bool, optional
-        Apply total-variation denoising separately for each channel. This
-        option should be true for color images, otherwise the denoising is
+        Whether to apply total variation denoising separately to each channel.
+        Should be True for color images, otherwise the denoising is
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:313]: Good question, but we haven't written "true" so far... We seem to have a mix of ``True`` and "True" in the docstrings; for example, in the same file:
https://github.com/scikit-image/scikit-image/blob/af45ceb3e9164d206789ecfb610d8ce31e3a1be3/skimage/filters/_fft_based.py#L25
and
https://github.com/scikit-image/scikit-image/blob/af45ceb3e9164d206789ecfb610d8ce31e3a1be3/skimage/filters/_fft_based.py#L108

[the remark this answers is not in the document]
Comment (lagru) [reply skimage/restoration/_denoise.py:313]: I'd just prefer the more explicit `True` which makes the actual value of the parameter crystal clear.

File: skimage/restoration/_denoise.py [hunk 4]
```diff
@@ -437,32 +465,50 @@ def _denoise_tv_chambolle_nd(image, weight=0.1, eps=2.e-4, max_num_iter=200):
...
+        fidelity to `image`).
     eps : float, optional
-        Relative difference of the value of the cost function that
-        determines the stop criterion. The algorithm stops when:
-
-            (E_(n-1) - E_n) < eps * E_0
-
+        Absolute value of relative difference of the cost function :math:`E`
+        that determines the stop criterion. The algorithm stops when
+        :math:`|E_{n-1} - E_n| < eps * E_0`.
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:502]: Thanks for double-checking! Do you think I might as well replace "eps" with "\varepsilon" (in order to render epsilon in LaTeX)? It would be the only place where "epsilon" appears as the Greek letter though...

File: skimage/restoration/_denoise.py [hunk 5]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
-
+        Stop criterion. The algorithm stops when
+        :math:`\sqrt{(u_n - u_{n-1})^2} < eps`.
     max_num_iter : int, optional
         Maximal number of iterations used for the optimization.
     isotropic : boolean, optional
         Switch between isotropic and anisotropic TV denoising.
     channel_axis : int or None, optional
-        If None, the image is assumed to be a grayscale (single channel) image.
+        If None, the image is assumed to be grayscale (single-channel).
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:305]: Does it mean that "NaN" here
https://github.com/scikit-image/scikit-image/blob/af45ceb3e9164d206789ecfb610d8ce31e3a1be3/skimage/segmentation/slic_superpixels.py#L207
should read ``Nan``?

File: skimage/restoration/_denoise.py [hunk 6]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
+        the smaller the `weight`, the more denoising (at
+        the expense of less similarity to `image`).
     eps : float, optional
-        Relative difference of the value of the cost function that determines
-        the stop criterion. The algorithm stops when::
-
-            SUM((u(n) - u(n-1))**2) < eps
-
+        Stop criterion. The algorithm stops when
+        :math:`\sqrt{(u_n - u_{n-1})^2} < eps`.
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:299]: You haven't missed any explanation, @lagru (and than you for your review, by the way). From the Goldstein and Osher reference, the lhs is the Euclidean norm of the difference between consecutive u's: This is what I intended to express with :math:`\sqrt{(u_n - u_{n-1})^2} < eps` but I was lazy and didn't include the sum over i's (vector components). There must be a normalization factor as well (I guess that what `total` is in the code): In a way, set "eps' = eps / N" and you're good (hehe)...

[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py]: @lagru how about 06b96ecd8ba766d2679701624eb90d40faa02a22?

File: skimage/restoration/_denoise.py [hunk 7]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
                        isotropic=True, *, channel_axis=None,
                        multichannel=False):
-    """Perform total-variation denoising using split-Bregman optimization.
+    r"""Perform total variation denoising using split-Bregman optimization.
 
-    Total-variation denoising (also know as total-variation regularization)
-    tries to find an image with less total-variation under the constraint
-    of being similar to the input image, which is controlled by the
-    regularization parameter ([1]_, [2]_, [3]_, [4]_).
+    Given :math:`f` a noisy image (input data),
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:269]: No, it is: "Given X a thing."

"an image $u$" is not the same thing as "given $u$ an image" :upside_down_face:

File: skimage/restoration/_denoise.py [hunk 8]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
+    total variation denoising (also known as total variation regularization)
+    aims to find an image :math:`u` with less total variation than :math:`f`,
+    under the constraint that :math:`u` remain similar to :math:`f`.
+    This can be expressed by the Rudin--Osher--Fatemi (ROF) minimization
+    problem:
+
+    .. math::
+
+        \min_{u} \sum_{i=0}^{N-1} \left( \left| \nabla{u_i} \right| + \frac{\lambda}{2}(f_i - u_i)^2 \right)
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:278]: Good question, I've definitely hesitated. So you think the higher-level stuff should be in the main description, and the technical details under Notes? I did it the other way around because we need to be specific when describing the parameters (cf. original issue).

File: skimage/restoration/_denoise.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:305]: I'm hesitant because our docstrings contain many more "If None" than "If ``None``" :thinking:

File: skimage/restoration/_denoise.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (grlee77) [reply skimage/restoration/_denoise.py:269]: Perhaps, but I think the suggestion does read more naturally

File: skimage/restoration/_denoise.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (grlee77) [reply skimage/restoration/_denoise.py:305]: I think it should likely be ``None``, but we just haven't done it very consistently so far. (same for ``True`` and ``False`` in many functions). At least, I think that is how the SciPy docs usually have them.

For NaN, it looks like SciPy uses ``nan``

**Review by grlee77 — DISMISSED** [review]
Thanks for taking this up @mkcor. It looks reasonable to me, having the same equation up top for both and explaining the difference in `weight` parameter under the `Parameters` section.

Please see just the remaining minor suggestions regarding ``nan`` and ``None``.

File: skimage/restoration/_denoise.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py:269]: There are two viable forumaltions: `given f, a noisy image (input data), ...` or `given a noisy image, f, ...`.  The latter versions reads more easily to me, but with the comma added the first is acceptable.

File: skimage/restoration/_denoise.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py:278]: I'd leave this in the description; it explains what the function does.

File: skimage/restoration/_denoise.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py:305]: This refers to the ``None`` object, so @lagru's suggestion is correct.

File: skimage/restoration/_denoise.py [hunk 9]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
+        If None, the image is assumed to be grayscale (single-channel).
         Otherwise, this parameter indicates which axis of the array corresponds
         to channels.
 
         .. versionadded:: 0.19
            ``channel_axis`` was added in 0.19.
     multichannel : bool, optional
-        Apply total-variation denoising separately for each channel. This
-        option should be true for color images, otherwise the denoising is
+        Whether to apply total variation denoising separately to each channel.
```
Comment (stefanv) [inline skimage/restoration/_denoise.py:312]:
```suggestion
        Whether or not to apply denoising separately to each channel.
```

File: skimage/restoration/_denoise.py [hunk 10]
```diff
@@ -264,41 +264,53 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
+        Should be ``True`` for color images, otherwise the denoising is
```
Comment (stefanv) [inline skimage/restoration/_denoise.py:313]:
```suggestion
        Should always be set to ``True`` for color images, otherwise the denoising is
```

File: skimage/restoration/_denoise.py [hunk 11]
```diff
@@ -307,16 +319,32 @@ def denoise_tv_bregman(image, weight=5.0, max_num_iter=100, eps=1e-3,
     u : ndarray
         Denoised image.
 
+    Notes
+    -----
+    Make sure to set the `channel_axis` parameter appropriately for color
```
Comment (stefanv) [inline skimage/restoration/_denoise.py:324]:
```suggestion
    Ensure that `channel_axis` parameter is set appropriately for color
```

File: skimage/restoration/_denoise.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py:468]: I like the old, more explicit wording here more. I think we use N-dimensional (capitalized) typically?

File: skimage/restoration/_denoise.py [hunk 12]
```diff
@@ -437,32 +465,50 @@ def _denoise_tv_chambolle_nd(image, weight=0.1, eps=2.e-4, max_num_iter=200):
...
+        Stop criterion :math:`\varepsilon > 0` (absolute value of relative
+        difference of the cost function :math:`E`).
+        The algorithm stops when :math:`|E_{n-1} - E_n| < \varepsilon * E_0`.
     max_num_iter : int, optional
         Maximal number of iterations used for the optimization.
     multichannel : bool, optional
-        Apply total-variation denoising separately for each channel. This
-        option should be true for color images, otherwise the denoising is
+        Whether to apply total variation denoising separately to each channel.
+        Should be ``True`` for color images, otherwise the denoising is
```
Comment (stefanv) [inline skimage/restoration/_denoise.py:507]:
```suggestion
        Set to ``True`` for color images, otherwise the denoising is
```

Comment (mkcor) [reply skimage/restoration/_denoise.py:507]: @stefanv I'm happy to apply the same suggestions as above ("Whether or not ..." and "Should always be set ...").

**Comment by mkcor** [comment]
Oops, c8adb416c10abf0e9378d57bd089497ca9097ba6 will conflict with #6583... @lagru

File: skimage/restoration/_denoise.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py:305]: I think that's perfectly fine too :)

File: skimage/restoration/_denoise.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:278]: Thanks for your input, @stefanv. Explicit is better than implicit!

File: skimage/restoration/_denoise.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py]: Ok. I've applied @lagru's suggestion here and elsewhere in the file: 9629d3a5a12dd9cfcc60836477b1287bd80939b3

File: skimage/restoration/_denoise.py (hunk 12 above)
Comment (mkcor) [reply skimage/restoration/_denoise.py]: Done: c8adb416c10abf0e9378d57bd089497ca9097ba6

File: skimage/restoration/_denoise.py [hunk 13]
```diff
@@ -264,59 +264,87 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
 
     Parameters
     ----------
     image : ndarray
-        Input data to be denoised (converted using img_as_float`).
-    weight : float
-        Denoising weight. The smaller the `weight`, the more denoising (at
-        the expense of less similarity to the `input`). The regularization
-        parameter `lambda` is chosen as `2 * weight`.
+        Input image to be denoised (converted using :ref:`~.img_as_float`).
```
Comment (mkcor) [inline skimage/restoration/_denoise.py:292]: @lagru apparently this causes a warning (treated as error):
```
skimage.restoration._denoise.denoise_tv_bregman:22:undefined label: '~.img_as_float'
```
Otherwise, the PR is ready for final review. Thanks!

**Review by stefanv — APPROVED** [review]

File: skimage/restoration/_denoise.py [hunk 14]
```diff
@@ -264,59 +264,87 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
         to channels.
 
         .. versionadded:: 0.19
            ``channel_axis`` was added in 0.19.
     multichannel : bool, optional
-        Apply total-variation denoising separately for each channel. This
-        option should be true for color images, otherwise the denoising is
-        also applied in the channels dimension. This argument is deprecated:
-        specify `channel_axis` instead.
+        Whether or not to apply denoising separately to each channel.
```
Comment (lagru) [inline skimage/restoration/_denoise.py:312]:
```suggestion
        Whether to apply denoising separately to each channel.
```

File: skimage/restoration/_denoise.py (hunk 13 above)
Comment (lagru) [reply skimage/restoration/_denoise.py:292]:
```suggestion
        Input image to be denoised (converted using :func:`~.img_as_float`).
```
`:func:` seems to work for some reason whereas `:ref:` doesn't, at least locally.

File: skimage/restoration/_denoise.py [hunk 15]
```diff
@@ -437,32 +465,50 @@ def _denoise_tv_chambolle_nd(image, weight=0.1, eps=2.e-4, max_num_iter=200):
...
 
     Parameters
     ----------
-    image : ndarray of ints, uints or floats
-        Input data to be denoised. `image` can be of any numeric type,
-        but it is cast into an ndarray of floats for the computation
-        of the denoised image.
+    image : ndarray
+        Input image to be denoised. If its dtype is not float, it gets
+        converted with :ref:`~.img_as_float`.
```
Comment (lagru) [inline skimage/restoration/_denoise.py:494]:
```suggestion
        converted with :func:`~.img_as_float`.
```

File: skimage/restoration/_denoise.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (lagru) [reply skimage/restoration/_denoise.py]: I think using `\varepsilon` would be "more proper". :smile: But in that case we should would make sure that the association `eps` - `\varepsilon` is mentioned before in the docstring.

File: skimage/restoration/_denoise.py (hunk 12 above)
Comment (lagru) [reply skimage/restoration/_denoise.py]: :smile: My IDE (I think LanguageTool) tells me 

> Consider shortening this phrase to just 'Whether'. It is correct though if you mean 'regardless of whether'.

File: skimage/restoration/_denoise.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (lagru) [reply skimage/restoration/_denoise.py]: See https://github.com/scikit-image/scikit-image/pull/6544#discussion_r1004201524.

Otherwise, $||u_n - u_{n-1}||_2$ looks like a clear improvement over the previous two versions! Though, I'm still skeptical that this represents the nesting of successive sums and roots as the norm is applied on the current difference `tx` + the previous `rmse`. But I don't have a better suggestion. :wink: 

Btw, I find it curious that Sphinx seems to render `\| x \|` as $|| x ||$ while GitHub renders it to $\| x \|$.

**Review by lagru — APPROVED** [review]
@mkor, feel welcome to address my remaining suggestion however you like (I may be confused about it). Apart from that it looks like you addressed everyone's concerns, including the blocking review by @grlee77. So feel free to go ahead and merge once you and the CI are ready. :wink: Thank you!

File: skimage/restoration/_denoise.py [hunk 16]
```diff
@@ -264,59 +264,87 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
...
+        Denoising weight. It is equal to :math:`\frac{\lambda}{2}`. Therefore,
+        the smaller the `weight`, the more denoising (at
+        the expense of less similarity to `image`).
     eps : float, optional
-        Relative difference of the value of the cost function that determines
-        the stop criterion. The algorithm stops when::
-
-            SUM((u(n) - u(n-1))**2) < eps
-
+        Stop criterion :math:`\varepsilon > 0`. The algorithm stops when
```
[the remark this answers is not in the document]
Comment (lagru) [reply skimage/restoration/_denoise.py:298]: > With "ε > 0" the reader understands / guesses that this parameter is 'small' and acts as an upper bound on an absolute value...

That's not at all how it reads to me. Let me try to explain my view point differently. "Stop criterion ε > 0" reads to me as "the stop criterion is ε > 0, this thing will stop if it is true" and not "ε is the stop criterion, it must be > 0". The former interpretation is obviously wrong with context, without context it's very misleading. I am not saying "Stop criterion ε > 0" is technically wrong, but I still maintain that it we might want to reword it to be less confusing to people such as me. :wink:

[the remark this answers is not in the document]
Comment (lagru) [reply skimage/restoration/_denoise.py:298]: Note that I don't want to bikeshed about this. If you feel that this is not an issue and my explanation doesn't make sense to you, I'm happy to leave it as is. :smile:

[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py:298]: `\varalpha` is not the criterion, but simply an argument to the criterion. The stop criterion is, as this stated before, that ``SUM((u(n) - u(n-1))**2) < eps``.  I'd just leave this section as is.

[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/restoration/_denoise.py:298]: @stefanv you are right, I have a used a metonymy... Should I replace "Stop criterion :math:`\varepsilon > 0`" with something like "Tolerance :math:`\varepsilon > 0` for stop criterion?"

@lagru no worries, I'm more than happy to have your perspective!

**Comment by mkcor** [comment]
@lagru thank you for your review!

File: skimage/restoration/_denoise.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (stefanv) [reply skimage/restoration/_denoise.py]: Tolerance works!

**Review by stefanv — APPROVED** [review]

**Comment by lagru** [comment]
I think CircleCI fails due to an invalid cache again like in #6527. Rendering in my local build looks fine and I think @grlee77's requrests were addressed. So merging. Thanks @mkcor! :tada:

**Comment by stefanv** [comment]
Thank you @mkcor !

**Comment by mkcor** [comment]
> I think CircleCI fails due to an invalid cache again like in #6527. Rendering in my local build looks fine and I think @grlee77's requests were addressed. So merging. Thanks @mkcor! tada

Thanks, @lagru! On my end, it seems that I couldn't merge because it was *technically* blocked by the 'requested changes.'
